### ISSUE TITLE
[cmake] test dependencies (Trac #1367)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1359]
Doing only `make icetray-test` is insufficient to build all the dependencies the test depends on. Building the entire metaproject solves this, so it's some internal dependency.

Migrated from https://code.icecube.wisc.edu/ticket/1367
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:03:13", 
    "description": "Doing only `make icetray-test` is insufficient to build all the dependencies the test depends on. Building the entire metaproject solves this, so it's some internal dependency.", 
    "reporter": "david.schultz", 
    "cc": "nega", 
    "resolution": "fixed", 
    "_ts": "1547251393300533", 
    "component": "cmake", 
    "summary": "[cmake] test dependencies", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-09-23T16:14:35", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150923T16:48:15: david.schultz commented:

> Building `icetray-pybindings` makes it work.  So I guess the cxx tests need the pybindings.

**Comment by icecube-trac** [comment]
Trac update at 20150923T17:38:29: 
* nega changed status from "new" to "accepted"
* nega changed owner from "david.schultz" to "nega"
* nega commented:

> These fixes should have been included with the ones that closed #695.

**Comment by icecube-trac** [comment]
Trac update at 20150923T17:40:50: nega commented:

> This issue results in an error like this:
> 
> 
> ```text
> [dladieu@cobalt04 build] ./env-shell.sh bin/icetray-test -sa
> terminate called after throwing an instance of 'boost::python::error_already_set'
> ./env-shell.sh: line 166:  2144 Aborted                 PATH=$_PATH LD_LIBRARY_PATH=$_LD_LIBRARY_PATH DYLD_LIBRARY_PATH=$_DYLD_LIBRARY_PATH PYTHONPATH=$_PYTHONPATH ICETRAY_CLASSPATH=$_ICETRAY_CLASSPATH I3_PLATFORM="" I3_SRC=$_I3_SRC I3_BUILD=$_I3_BUILD I3_PORTS=$_I3_PORTS I3_TESTDATA=$_I3_TESTDATA ROOTSYS=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/root-v5.34.18 I3_SHELL=$_I3_SHELL G4LEVELGAMMADATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/PhotonEvaporation2.2 G4PIIDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/G4PII1.3 G4NEUTRONXSDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/G4NEUTRONXS1.1 G4ABLADATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/G4ABLA3.0 G4RADIOACTIVEDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/RadioactiveDecay3.4 G4LEDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/G4EMLOW6.23 G4NEUTRONHPDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/G4NDL4.0 G4REALSURFACEDATA=/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/i3ports/share/geant4/data/Geant4-9.5.0/data/RealSurface1.0 $NEW_SHELL $ARGV
> [dladieu@cobalt04 build] 
> 
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150923T17:43:02: 
* nega commented:

> In changeset 2319:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2319"
> make icetray-pybindings a dependency of all EXE tests
> 
> additionally, add a USE_PYBINDINGS to i3_test_executable(). This allows
> projects to specify dependent pybindings, so that they'll be automatically
> be built.
> 
> closes #1359
> ```


* nega changed status from "accepted" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:10: 
* olivas changed milestone from "offline V15-12-00" to ""
* olivas commented:

> Milestone offline V15-12-00 deleted

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:03:13: nwhitehorn commented:

> In changeset 170133:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170133"
> make icetray-pybindings a dependency of all EXE tests
> 
> additionally, add a USE_PYBINDINGS to i3_test_executable(). This allows
> projects to specify dependent pybindings, so that they'll be automatically
> be built.
> 
> closes #1359
> ```
